Add new extension: dlmoji

Hydrapse opened the pull request:
## Description

An intelligent Emoji Dictionary helps to get related Emojis, Phrases, and AI-powered Translations in Raycast.
### Features
1. Supports Chinese & English Emoji Search
2. Supports Semantic, Verbatim, or Blended Emoji Translation
3. Gets related Emojis 🧐 or Emoji Phrases  🛌🏃‍♀️ with Explanations

## Screencast

<img width="1000" alt="dlmoji-1" src="https://user-images.githubusercontent.com/45584876/179364432-11eb2a78-4904-4236-a6b4-47f2c3cbace4.png">

<img width="1000" alt="dlmoji-3" src="https://user-images.githubusercontent.com/45584876/179364439-0a31a470-14d6-4e32-81ed-6ae8f33210a2.png">

![Config-Panel](https://user-images.githubusercontent.com/45584876/179364513-b6e7c361-bf3b-4b17-9e04-1ba15ddf34ba.png)

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

Hydrapse commented:
Hi @danielsequeira,
Sorry for bothering you. I'm wondering if something goes wrong, or something I can do to progress the review.
Looking forward to your reply!

danielsequeira commented:
@Hydrapse one question, why the name DLmoji?

danielsequeira commented:
On another note, I think we will need to improve the UX a bit on the extension. In cases where the emoji description is too long, it doesn't look good to render it in the action panel and there are cases where even the action panel doesn't have enough space for it:
![dlmoji 2022-07-27 at 13 28 09](https://user-images.githubusercontent.com/12091319/181237223-00a9bff0-ba7d-444c-9c8e-fcee22c5fcb5.png)

I would suggest for you to use our detail component: https://developers.raycast.com/api-reference/user-interface/detail where we could have a detail section for each emoji and allow the entire description to be rendered.

Hydrapse commented:
> @Hydrapse one question, why the name DLmoji?

That's because I follow the nomenclature of [deepmoji](https://github.com/bfelbo/DeepMoji).
DLmoji means emoji search powered by Deep Learning or comparable intelligent APIs, which can handle phrase or sentence level queries, different from the existing [Emoji Search](https://www.raycast.com/FezVrasta/emoji) extension.

danielsequeira commented:
> > @Hydrapse one question, why the name DLmoji?
> 
> That's because I follow the nomenclature of [deepmoji](https://github.com/bfelbo/DeepMoji). DLmoji means emoji search powered by Deep Learning or comparable intelligent APIs, which can handle phrase or sentence level queries, different from the existing [Emoji Search](https://www.raycast.com/FezVrasta/emoji) extension.

Got you! Thanks for the explanation. I think in that case it would probably make sense to mention deep learning in the package.json description and in the readme. I can imagine a lot of people like myself not really knowing what it means and that can affect the searches in the store for this extension

Hydrapse commented:
ok, I got it. I'll update the description and try to improve the UX by following the advice 😉

Hydrapse commented:
Sorry for being late. I've added the Detail Page for Emoji description and updated the Readme & package.json description. 
Please have a look. Thanks.
<img width="862" alt="image" src="https://user-images.githubusercontent.com/45584876/185074061-d497980e-72d8-4f28-8edf-c89f7811f9ed.png">

danielsequeira commented on extensions/dlmoji/src/dlmoji.tsx:
```diff
@@ -0,0 +1,175 @@
+            let enText: string = queryText
+            if (hasChinese) {
+                // const res = await fetchDeepl(queryText)
+                // enText = res.data.data
```
please remove commented code

danielsequeira commented on extensions/dlmoji/src/api.tsx:
```diff
@@ -0,0 +1,145 @@
+
+    // let data = new FormData();
+    // data.append('keywords', queryText)
+    // data.append('lang', lang)
```
please remove commented code

danielsequeira commented on extensions/dlmoji/src/api.tsx:
```diff
@@ -0,0 +1,145 @@
+            }),
+            {
+                headers: {
+                    // 'Content-Type': 'multipart/form-data',
```
please remove commented code

danielsequeira commented on extensions/dlmoji/src/TranslateResult.tsx:
```diff
@@ -0,0 +1,163 @@
+
+    return finalData.map((text, idx) => {
+        return {
+            // title: finalData.length - 1 === idx && idx > 0 ? "All" : truncate(text),
```
please remove commented code

danielsequeira commented:
@Hydrapse thanks for the changes, I just added a few more comments, once they are addressed we should be good to merge this.

Hydrapse commented:
Thanks @danielsequeira. I've committed all your changes.

danielsequeira commented:
@Hydrapse thanks for the changes. One last thing I forgot, since it is not possible to read the entire description while in the List view and because you now have the View Description option, I think it makes sense to remove the description text from the list items:
![dlmoji 2022-08-22 at 13 43 20](https://user-images.githubusercontent.com/12091319/185913621-470b7fff-d12b-420c-8f76-fec16db6d4e0.png)

Hydrapse commented:
The API service of "[deepmoji](https://github.com/bfelbo/DeepMoji) Emotion Analysis" is right here (**for Test Only**): 

`http://106.55.173.219:18878/ `

I think it might benefit the final testing 😊.

<img width="314" alt="image" src="https://user-images.githubusercontent.com/45584876/186088099-6a2565ff-0ec3-4082-8c44-6dbf7340f175.png">

danielsequeira commented:
All looks good @Hydrapse, I just noticed though that now that you removed the description from the list items, your screenshots are not correct, because they still show the description and this can confuse users. Could you update them? If you do it in the next few hours, we can still merge and publish this today 🙂

Hydrapse commented:
Besides updating all the screenshots in README and metadata, I also add a new feature.

I've found another Emoji Translation API, which is a perfect substitute for the "18dao Chinese Verbatim" API. Then I merge this new API into the Emoji Translation Section, which works for "Chinese To Emoji" Translation and "Emoji To Text" Translation:
<img width="862" alt="image" src="https://user-images.githubusercontent.com/45584876/186398877-84d85bd5-a6e9-489c-9bbb-e4156cef4026.png">
<img width="862" alt="image" src="https://user-images.githubusercontent.com/45584876/186398932-5269a95e-680b-4600-ab76-1fd3df6cb140.png">
